### PR TITLE
Added settings page

### PR DESCRIPTION
**Pull request by fertandil87**
https://globalfishingwatch.atlassian.net/browse/VH-74

The translations in another PR
The options for the multiselects need to be defined
image

image

Also added a new feature in the multiselect component @j8seangel

### CODE REVIEW

File: applications/vessel-history/src/App.tsx [hunk 1]
```diff
@@ -4,11 +4,12 @@ import GFWAPI from '@globalfishingwatch/api-client'
 import useGFWLogin from '@globalfishingwatch/react-hooks/dist/use-login'
 import Home from 'features/home/Home'
 import { getLocationType } from 'routes/routes.selectors'
-import { PROFILE } from 'routes/routes'
+import { PROFILE, SETTINGS } from 'routes/routes'
 import { BASE_URL, SPLASH_TIMEOUT } from 'data/constants'
 import Profile from 'features/profile/Profile'
 import Splash from 'features/splash/Splash'
 import './App.css'
+import Settings from 'features/settings/Settings'
```
Comment (rdgfuentes) [inline applications/vessel-history/src/App.tsx:12]: This import should be on line 11 before importing the App.css

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+  const eezs = EEZ_OPTIONS.filter((option) => settings.eezs?.includes(option.id))
+  const rfmos = RFMOS_OPTIONS.filter((option) => settings.rfmos?.includes(option.id))
+  const mpas = EEZ_OPTIONS.filter((option) => settings.mpas?.includes(option.id))
+
+  return (
+    <div className={styles.settingsFieldsContainer}>
+      <div className={styles.settingsField}>
+        <label>
+          Events in There EEZs
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/Encounters.tsx:72]: Typo, it should say 'Events in these EEZs'

File: applications/vessel-history/src/features/settings/Settings.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+          {options.map((option: string) => (
+            <li key={option} onClick={() => setSelectedOption(option)}>
+              {optionsData[option].title}
+              <IconButton type="default" size="default" icon="arrow-right" />
+            </li>
+          ))}
+        </ul>
+      )}
+      {selectedOption === 'fishing_events' && (
+        <FishingEvents settings={settings.fishingEvents} section="fishingEvents"></FishingEvents>
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/Settings.tsx:73]: 👏

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  settings: SettingsEvents
+  section: string
+}
+
+const Encounters: React.FC<SettingsProps> = (props): React.ReactElement => {
+  const { settings, section } = props
+
+  const { setSettingOptions, setSetting } = useApplySettingsConnect()
+
+  const EEZ_OPTIONS: SelectOption[] = [
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/Encounters.tsx:22]: We should get or build this list dynamically.

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+      id: 'ghana',
+      label: 'Ghana',
+    },
+    {
+      id: 'argentina',
+      label: 'Argentina',
+    },
+  ]
+
+  const RFMOS_OPTIONS: MultiSelectOption[] = [
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/Encounters.tsx:33]: This should also be built dynamically

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+  const eezs = EEZ_OPTIONS.filter((option) => settings.eezs?.includes(option.id))
+  const rfmos = RFMOS_OPTIONS.filter((option) => settings.rfmos?.includes(option.id))
+  const mpas = EEZ_OPTIONS.filter((option) => settings.mpas?.includes(option.id))
+
+  return (
+    <div className={styles.settingsFieldsContainer}>
+      <div className={styles.settingsField}>
+        <label>
+          Events in There EEZs
+          <IconButton type="default" size="tiny" icon="info" tooltip="testing tootip"></IconButton>
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/Encounters.tsx:73]: `tooltip="testing tootip"` ? 😅

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+interface SettingsProps {
+  settings: SettingsEvents
+  section: string
+}
+
+const Encounters: React.FC<SettingsProps> = (props): React.ReactElement => {
+  const { settings, section } = props
+
+  const { setSettingOptions, setSetting } = useApplySettingsConnect()
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/Encounters.tsx:20]: 👌

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    },
+    {
+      id: 'WCPFC',
+      label: 'WCPFC',
+    },
+  ]
+
+  const eezs = EEZ_OPTIONS.filter((option) => settings.eezs?.includes(option.id))
+  const rfmos = RFMOS_OPTIONS.filter((option) => settings.rfmos?.includes(option.id))
+  const mpas = EEZ_OPTIONS.filter((option) => settings.mpas?.includes(option.id))
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/Encounters.tsx:66]: MPAs are not EEZ. We should get the actual list of MPAs

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+      <div className={cx(styles.settingsField, styles.inlineRow)}>
+        <label>
+          DURATION
+          <IconButton type="default" size="tiny" icon="info" tooltip="testing tootip"></IconButton>
+        </label>
+        <span>Longer than</span>
+        <InputText
+          type="number"
+          value={settings.duration ?? 2}
+          min={2}
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/Encounters.tsx:143]: Why 2 is the min? As far as I could read in the ticket it should be 1
> Range to select goes from 1 hour to 99 hours.
Use constants for these ranges too.

File: applications/vessel-history/src/features/settings/components/SettingsComponents.module.css [hunk 10]
```diff
@@ -0,0 +1,24 @@
+.settingsFieldsContainer {
+  margin: 15px;
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/SettingsComponents.module.css:2]: Try to use the variables that we currently have defined in our [base.css](https://github.com/GlobalFishingWatch/frontend/blob/develop/packages/ui-components/src/base.css#L59) for spacing and sizes whenever is possible:
```

  /* SIZES */
  --size-XS: 2rem;
  --size-S: 2.8rem;
  --size-M: 4rem;
  --size-L: 6rem;
  --space-XS: 0.5rem;
  --space-S: 1rem;
  --space-M: 2rem;
  --space-L: 4rem;
```

File: packages/ui-components/src/multi-select/MultiSelect.tsx [hunk 11]
```diff
@@ -18,6 +18,7 @@ import { MultiSelectOption, MultiSelectOnChange } from './index'
 interface MultiSelectProps {
   label?: string
   placeholder?: string
+  placeholderDisplayAll?: boolean
```
Comment (rdgfuentes) [inline packages/ui-components/src/multi-select/MultiSelect.tsx:21]: 👏  Nice feature

**Review by rdgfuentes — CHANGES_REQUESTED** [review]
Overall looks good, the only concern I have is that there is too much repeated code in `Encounters`, `FishingEvents` and `LoiteringEvents` components. 
Instead of repeating the same logic in every component I'd prefer to:
* Extract and parametrize (min, max, setting key, etc..) that component behavior into a single component, `ActivityEvents` might be an appropriate name for it.
* Rename `Encounters` to `EncounterEvents` to follow the same convention
* Inside `EncounterEvents`, `FishingEvents` and `LoiteringEvents` just render the `ActivityEvents` component with its appropriate properties.

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 4 above)
Comment (fertandil87) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:22]: we know that, when I know where to take the list I will do, also I don't know if the different settings use the same EEZ, so I created 3 different list por the moment

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 2 above)
Comment (fertandil87) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:72]: will be fixed in the translation

File: applications/vessel-history/src/features/settings/components/Encounters.tsx [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+      </div>
+      <div className={cx(styles.settingsField, styles.inlineRow)}>
+        <label>
+          DURATION
+          <IconButton type="default" size="tiny" icon="info" tooltip="testing tootip"></IconButton>
+        </label>
+        <span>Longer than</span>
+        <InputText
+          type="number"
+          value={settings.duration ?? 2}
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:142]: is not magic, is the min value supported for this field

File: applications/vessel-history/src/offline-store.ts [hunk 13]
```diff
@@ -3,6 +3,7 @@ import { OfflineVessel } from 'types/vessel'
 
 export class VesselHistoryIdb extends Dexie {
   vessels: Dexie.Table<OfflineVessel, string>
+  settings: Dexie.Table<any>
```
[the remark this answers is not in the document]
Comment (fertandil87) [reply applications/vessel-history/src/offline-store.ts:6]: I forget to remove this

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 8 above)
Comment (fertandil87) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:66]: we don't have the lists yet

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 6 above)
Comment (fertandil87) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:73]: just adding the tooltip while I wait for the final help tests

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 5 above)
Comment (fertandil87) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:33]: yes, when I have the final lists..

File: applications/vessel-history/src/features/settings/components/SettingsComponents.module.css (hunk 10 above)
Comment (fertandil87) [reply applications/vessel-history/src/features/settings/components/SettingsComponents.module.css:2]: So I need to modify the design? XS is 20px

**Comment by fertandil87** [comment]
> * name `Encounters` to `EncounterEvents` to follow the sam

I thought the same for that I merged the save settings, but I don't what data is comming to fill the selectors and also I am not really sure if they will be equal, now looks that they just old copy the same inputs from fishing events, and if then they are now equal I need to rollback evething. But if you are sure that the multiselect and labels will be the same after we know what lists use in each case let me know and I will do it

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (rdgfuentes) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:142]: Magic number refers to unnamed constant values in the code (https://en.wikipedia.org/wiki/Magic_number_(programming)), it's better to assign them to constants for clarity and maintainability

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 4 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:22]: I doubt they would differ for different event types. Beyond that we could define where in our state we are planning on storing all these lists using these values as initialState and adding a comment to be completed/implemented later, with that we can define the selector to retrieve it and also use it here.

File: applications/vessel-history/src/features/settings/components/Encounters.tsx (hunk 8 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/features/settings/components/Encounters.tsx:66]: However we can resolve how is going to get that list. Same as https://github.com/GlobalFishingWatch/frontend/pull/533#discussion_r635510671

File: applications/vessel-history/src/features/settings/components/SettingsComponents.module.css (hunk 10 above)
Comment (rdgfuentes) [reply applications/vessel-history/src/features/settings/components/SettingsComponents.module.css:2]: No, I'm suggestion to use the variables from the design that best match your needs. Juanca @satellitestudiodesign what do you think?

**Comment by rdgfuentes** [comment]
> > * name `Encounters` to `EncounterEvents` to follow the sam
> 
> I thought the same for that I merged the save settings, but I don't what data is comming to fill the selectors and also I am not really sure if they will be equal, now looks that they just old copy the same inputs from fishing events, and if then they are now equal I need to rollback evething. But if you are sure that the multiselect and labels will be the same after we know what lists use in each case let me know and I will do it

I'm pretty sure they will be the same for the three type of events, but asked Gise to confirm it and cc'ed you in the slack thread. Ping her for confirmation if the reply does not arrive soon and if ok let's do it.

File: applications/vessel-history/src/features/settings/components/ActivityEvents.tsx [hunk 14]
```diff
@@ -1,22 +1,25 @@
...
+  maxDuration: number
+  minDistance: number
+  maxDistance: number
 }
 
-const Encounters: React.FC<SettingsProps> = (props): React.ReactElement => {
+const ActivityEvents: React.FC<SettingsProps> = (props): React.ReactElement => {
   const { settings, section } = props
-
+  const { t } = useTranslation()
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/ActivityEvents.tsx:22]: 👍

File: applications/vessel-history/src/features/settings/components/ActivityEvents.tsx [hunk 15]
```diff
@@ -128,35 +141,35 @@ const Encounters: React.FC<SettingsProps> = (props): React.ReactElement => {
...
         </label>
-        <span>Longer than</span>
+        <span>{t('settings.longerThan',"Longer than")}</span>
         <InputText
           type="number"
-          value={settings.duration ?? 2}
-          min={2}
-          max={99}
+          value={settings.duration ?? 0}
+          min={props.minDuration}
```
Comment (rdgfuentes) [inline applications/vessel-history/src/features/settings/components/ActivityEvents.tsx:156]: 👍

**Review by rdgfuentes — APPROVED** [review]
Thanks @fertandil87 looks good, the only thing I'd change is adding the `
import '@globalfishingwatch/ui-components/dist/base.css'` in the index.tsx as we did on fishing-map so it's not necessary to import it in every component we want to use the variables.

**Pull request merged**